controllers/user.controller.js

```diff
@@ -0,0 +1,106 @@
+const userService = require("../services/user.service");
+
+const {
+  isValidEmailSyntax,
+  isValidPassword,
+  composite,
+  compare,
+  getRandString,
+  getPublicUserData,
+  respondWithError,
+  respondWithStatus,
+} = require("../services/utils");
+
+const bcrypt = require("bcrypt");
+
+const { SERCURITY, ERRORS, CRUD_OPS } = require("../constants");
+
+const register = async (req, res) => {
+  try {
+    const { name, email, password } = req.body;
+    if (!isValidEmailSyntax(email)) {
+      return respondWithError(res, ERRORS.INVALID_EMAIL_ADDRESS);
+    }
+    if (!isValidPassword(password)) {
+      return respondWithError(res, ERRORS.INVALID_PASSWORD);
+    }
+    if (await userService.isExistingUser(name)) {
+      return respondWithError(res, ERRORS.NAME_OR_EMAIL_ERROR);
```

Comment:
Do you think that `isUserExist` sounds better?

Reply:
Yes.

Reply:
fixed in new commit, will publish soon